plugins/modules/purefa_network.py

```diff
@@ -227,35 +266,106 @@ def update_fc_interface(module, array, interface, api_version):
+                    msg="Child subinterface {0} does not exist".format(inter)
+                )
+            if init:
+                subinterfaces.append(FixedReferenceNoId(name="ct0." + inter))
+            else:
+                subinterfaces.append("ct0." + inter)
+            if not purity_vm:
+                if init:
+                    subinterfaces.append(FixedReferenceNoId(name="ct1." + inter))
+                else:
+                    subinterfaces.append("ct1." + inter)
+    return subinterfaces
+
+
 def update_interface(module, array, interface):
     """Modify Interface settings"""
     changed = False
     current_state = {
+        "enabled": interface["enabled"],
         "mtu": interface["mtu"],
         "gateway": interface["gateway"],
         "address": interface["address"],
         "netmask": interface["netmask"],
         "services": sorted(interface["services"]),
+        "slaves": sorted(interface["slaves"]),
     }
+    array6 = get_array(module)
+    subinterfaces = current_state["slaves"]
+    if module.params["subinterfaces"]:
+        if sorted(module.params["subinterfaces"]) != current_state["slaves"]:
+            subinterfaces = _create_subinterfaces(module, array6)
+        else:
+            subinterfaces = current_state["slaves"]
+    if module.params["subordinates"]:
+        if sorted(module.params["subordinates"]) != current_state["slaves"]:
+            subinterfaces = _create_subordinates(module, array6)
+        else:
+            subinterfaces = current_state["slaves"]
+    if module.params["enabled"] != current_state["enabled"]:
+        enabled = module.params["enabled"]
+    else:
+        enabled = current_state["enabled"]
+    if not current_state["gateway"]:
+        if valid_ipv4(interface["address"]):
+            current_state["gateway"] = None
+        elif valid_ipv6(interface["address"]):
+            current_state["gateway"] = None
     if not module.params["servicelist"]:
         services = sorted(interface["services"])
     else:
         services = sorted(module.params["servicelist"])
     if not module.params["address"]:
         address = interface["address"]
+        netmask = interface["netmask"]
     else:
-        if module.params["gateway"] and module.params["gateway"] != "0.0.0.0":
-            if module.params["gateway"] and module.params["gateway"] not in IPNetwork(
-                module.params["address"]
-            ):
+        if module.params["gateway"] and module.params["gateway"] not in [
+            "0.0.0.0",
+            "::",
+        ]:
+            if module.params["gateway"] not in IPNetwork(module.params["address"]):
                 module.fail_json(msg="Gateway and subnet are not compatible.")
             elif not module.params["gateway"] and interface["gateway"] not in [
                 None,
                 IPNetwork(module.params["address"]),
             ]:
                 module.fail_json(msg="Gateway and subnet are not compatible.")
```

Comment:
This elif statement is never equating to true. The preceding if statement on line 355, one of the checks is if `module.params["gateway"]` exists so the latter check on line 361 will always be false. 

I believe the highlighted section should be un-indented once so the elif statement is a part of the if statement on line 355 and not within that if statement.

Reply:
Fixed